### DISCUSSION TITLE
Inline assembly

### DISCUSSION THREAD

**Discussion by mmcheraghi** [stnolting/neorv32#910] [General]
Hi
Does anyone know a good source to learn writing inline assembly in C for RISC-V?
My goal is to access memory locations in a C code.
Thanks

**Comment by stnolting** [comment]
The NEORV32 software framework already provides predefined functions (using inline assembly) for reading/writing from/to arbitrary memory addresses, e.g.:

https://github.com/stnolting/neorv32/blob/c5dd61755215b36af7f9061af0393ffbeb95fde8/sw/lib/include/neorv32_cpu.h#L159-L165

For a nice RISC-V assembly reference I can recommend the RISC-V Assembly Programmer's Manual: https://github.com/riscv-non-isa/riscv-asm-manual/blob/main/riscv-asm.md

For the C part I would refer to the GCC inline assembly documentation: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html

**Comment by mmcheraghi** [comment]
Thanks a lot. It was very helpful🙏
